File: app/src/main/java/com/musicplayer/SocyMusic/data/base/AppDatabase.java
package com.musicplayer.SocyMusic.data.base;

import androidx.room.Database;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import com.musicplayer.SocyMusic.data.Album;
import com.musicplayer.SocyMusic.data.Playlist;
import com.musicplayer.SocyMusic.data.Song;

@Database(entities = {Song.class, Playlist.class, Album.class, PlaylistSong.class}, version = 2)
@TypeConverters({FileConverter.class, UUIDConverter.class})
public abstract class AppDatabase extends RoomDatabase {
    public static final String DATABASE_NAME = "socyMusic.sqlite3";

    public abstract SongDao songDao();

    public abstract PlaylistDao playlistDao();

    public abstract PlaylistSongDao playlistSongDao();

    public abstract AlbumDao albumDao();

}

File: app/src/main/java/com/musicplayer/OpenMusic/data/base/AppDatabase.kt
package com.musicplayer.OpenMusic.data.base

import androidx.room.Database
import com.musicplayer.OpenMusic.data.Song
import com.musicplayer.OpenMusic.data.Playlist
import com.musicplayer.OpenMusic.data.Album
import androidx.room.TypeConverters
import androidx.room.RoomDatabase

@Database(entities = [Song::class, Playlist::class, Album::class, PlaylistSong::class], version = 2)
@TypeConverters(FileConverter::class, UUIDConverter::class)
abstract class AppDatabase : RoomDatabase() {
    abstract fun songDao(): SongDao
    abstract fun playlistDao(): PlaylistDao
    abstract fun playlistSongDao(): PlaylistSongDao
    abstract fun albumDao(): AlbumDao

    companion object {
        const val DATABASE_NAME = "OpenMusic.sqlite3"
    }
}